### PR TITLE
Hescore 1586

### PR DESCRIPTION
**Pull request by plithnar**
Will tag new version following merge so we can update GUI and API project dependencies.

@thepriestscottage , I wasn't sure who all to put as reviewer or who to assign it to. Feel free to add more as is appropriate.

### CODE REVIEW

File: home_audit.node.js [hunk 1]
```diff
@@ -1306,6 +1306,23 @@ let validationRules = {
             );
         }
     },
+
+
+    _is_servicing_duct_system: function(value, sys, duct) {
```
Comment (thepriestscottage) [inline home_audit.node.js:1311]: Can you add a comment explaining what this does?

**Review by thepriestscottage — CHANGES_REQUESTED** [review]
This is probably my fault for failing to explain, but we actually no longer use branch names for hes-validation-engine versioning, but tags.  So, this should be pointed at `master`, and `release` (tragically) was 75 commits behind, so you will need to merge and resolve conflicts. I am sorry :(

File: home_audit.node.js [hunk 2]
```diff
@@ -1306,6 +1306,23 @@ let validationRules = {
...
+        let ductTypes = ['central_furnace', 'heat_pump', 'gchp', 'split_dx'];
+        if (ductTypes.indexOf(_homeValues['heating_type_'+sys]) !== -1  &&
+            ductTypes.indexOf(_homeValues['cooling_type_'+sys]) !== -1)
+        {
+            if(!TypeRules._is_empty(value)) {
+                return new Validation(
+                    'Values may not be defined for system that does not call for ducts. Please only set values for ducts on Central Furnace, Heat Pump, Ground Coupled Heating Pump, or Central Air Conditioner systems',
+                    ERROR
+                )
+                }
```
[the remark this answers is not in the document]
Comment (plithnar) [reply home_audit.node.js:1472]: Done

File: home_audit.node.js [hunk 3]
```diff
@@ -1306,6 +1306,23 @@ let validationRules = {
...
         }
     },
+
+
+    _is_servicing_duct_system: function(value, sys, duct) {
+        // If heating/cooling system does not require ducts, we need to ensure the
+        // user has not entered values for ducts
+        let ductTypes = ['central_furnace', 'heat_pump', 'gchp', 'split_dx'];
+        if (ductTypes.indexOf(_homeValues['heating_type_'+sys]) !== -1  &&
+            ductTypes.indexOf(_homeValues['cooling_type_'+sys]) !== -1)
```
[the remark this answers is not in the document]
Comment (plithnar) [reply home_audit.node.js:1465]: Swap made

File: home_audit.node.js [hunk 4]
```diff
@@ -1306,6 +1306,23 @@ let validationRules = {
...
+        {
+            if(!TypeRules._is_empty(value)) {
```
[the remark this answers is not in the document]
Comment (plithnar) [reply home_audit.node.js:1467]: Yeah, I was just not wanting the line to be overly long. Collapsed lines

File: home_audit.node.js (hunk 1 above)
Comment (plithnar) [reply home_audit.node.js:1460]: JSDoc comment added to signature.

File: home_audit.node.js [hunk 5]
```diff
@@ -1306,6 +1306,23 @@ let validationRules = {
             );
         }
     },
+
+
+    _is_servicing_duct_system: function(value, sys, duct) {
+        // If heating/cooling system does not require ducts, we need to ensure the
+        // user has not entered values for ducts
+        let ductTypes = ['central_furnace', 'heat_pump', 'gchp', 'split_dx'];
```
[the remark this answers is not in the document]
Comment (plithnar) [reply home_audit.node.js:1463]: Unfortunately, there isn't a spot where 1) the duct types are defined individually or 2) the above list is defined as "ductless" duct types. Is something I will look to implement in 1573 with the other validation changes

**Review by thepriestscottage — APPROVED** [review]

**Pull request merged**